### ISSUE TITLE
empty_enum_arguments is triggered for pattern matching using functions

### ISSUE THREAD

**Issue by bsarrazin** [realm/SwiftLint#1615]
```swift
switch className {
case ViewControllerA.someFunctionReturningString(): // triggered
    return valueA
case ViewControllerB.someFunctionReturningString(): // triggered
    return valueB
case ViewControllerC.someFunctionReturningString(): // triggered
    return valueC
default:
    return nil
}
```

**Comment by marcelofabri** [comment]
Can you try with `master`? This should be fixed by #1598.

**Comment by marcelofabri** [comment]
I'm closing this. Feel free to reopen if #1598 doesn't fix it.

**Comment by bsarrazin** [comment]
@marcelofabri sorry for not getting back sooner.
I have not tried `master`. I trust that #1598 fixes, will wait until the next release to validate it.
I opened the bug because I couldn't find anything with `empty_enum_arguments`. Sorry for duplicating the bug.

**Comment by marcelofabri** [comment]
@bsarrazin No worries at all. Feel free to reopen if the next release doesn't fix this issue.
